### ISSUE TITLE
Upgrading to 0.96.0 (node behind a proxy) breaks the app because of the API Url Scheme

### ISSUE THREAD

**Issue by jpbourgeon** [nocodb/nocodb#3536]
Hi

I upgraded to 0.96.0 today and I cannot signin normally anymore.

NocoDB is installed behind an Nginx reverse proxy that enforces domain name and SSL certificate.

The app loads properly, however the login process throws a network error because the API url is not served under https.

I tried various combinations of  `NC_BASEURL_INTERNAL` and `NC_PUBLIC_URL` with no luck so far :

- removing the scheme : `localhost:$PORT`
- enforcing the scheme : `https://localhost:$PORT`

Could you help me with setting up the right parameters for my instance ?

**Project info**

```
NocoDB used as docker : false
NocoDB version : 0.96.0
Database used in NC_DB URL  : mysql 
Database version : Ver 15.1 Distrib 10.3.34-MariaDB, for Database version
Project was created  by clicking : New Project
OS on which NocoDB is running : debian-linux-gnu (x86_64)
Node.js version if running as node : v12.22.12
```

**Screenshots**

![image](https://user-images.githubusercontent.com/3356641/188874590-f151068b-0034-4df5-9d2f-2993bddf87ca.png)

** Related issues **

- #3239

**Comment by jpbourgeon** [comment]
Here is the location part in my nginx config
```
location ^~ / {
        if ($scheme != "https") {
                return 301 https://$server_name$request_uri;
        }
        proxy_set_header X-Forwarded-For $remote_addr;
        proxy_set_header X-Forwarded-Proto 'https';
        proxy_set_header Host $host;
        proxy_pass      http://192.168.1.175:8080;
    }
```

However I think this is an issue raised by the browser before getting a chance to go to the api since the scheme is hardcoded.

**Comment by o1lab** [comment]
See if this below build helps ? 

`nocodb/nocodb-timely:0.96.0-pr-3528-20220907-0904`

We're putting this out now

**Comment by jpbourgeon** [comment]
Thanks @o1lab for your prompt answer !
I patched up to 0.96.1 and now I can connect and for what I can tell, I can now signin in a use nocodb properly on my side.

The error still shows in the browser console concerning websockets, but it is non blocking so far.

```
Mixed Content: The page at 'https://nocodb.cdg71.fr/dashboard/#/signin'
was loaded over HTTPS, but requested an insecure XMLHttpRequest endpoint 
'http://nocodb.cdg71.fr/socket.io/?EIO=4&transport=polling&t=OCO7cnw'.
This request has been blocked; the content must be served over HTTPS.
```

**Comment by o1lab** [comment]
Ok cool.
